cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -346,11 +346,42 @@ pub mod CometClientComponent {
 
             let header_height = header.clone().signed_header.height;
 
-            // TODO: Implement consensus state pruning mechanism.
+            let update_heights = self.read_update_heights(client_sequence);
+            let mut update_heights_span = update_heights.span();
 
-            if !self.consensus_state_exists(client_sequence, header_height.clone()) {
-                let mut client_state = self.read_client_state(client_sequence);
+            let mut client_state = self.read_client_state(client_sequence);
 
+            let mut heights_kept = array![];
+            let mut check_in_progress = true;
+            // Since th Heights are sorted when stored, as soon as we find the first
+            // Height which isn't expired we can stop checking the rest and build
+            // the new Heights array which are kept.
+            while let Option::Some(height) = update_heights_span.pop_front() {
+                if check_in_progress {
+                    let consensus_state = self
+                        .read_consensus_state(client_sequence, height.clone());
+
+                    let expiry_timestamp = consensus_state.timestamp.as_secs()
+                        + client_state.trusting_period.seconds;
+
+                    let host_timestamp = get_block_timestamp();
+
+                    if expiry_timestamp <= host_timestamp {
+                        let consensus_zero = CometConsensusStateImpl::zero();
+                        self.write_consensus_state(client_sequence, height.clone(), consensus_zero);
```

Comment:
I didn't find a way to delete a Starknet storage `Map` entry.
Should the `Map` be rebuilt without the entry? @rnbguy

Reply:
From https://book.cairo-lang.org/ch03-02-dictionaries.html#basic-use-of-dictionaries

> Before heading on and explaining how dictionaries are implemented it is worth mentioning that once you instantiate a Felt252Dict<T>, behind the scenes all keys have their associated values initialized as zero. This means that if for example, you tried to get the balance of an inexistent user you will get 0 instead of an error or an undefined value. This also means there is no way to delete data from a dictionary. Something to take into account when incorporating this structure into your code.

I guess we can't delete a data. We just set it to `Zero::zero` value.